### DISCUSSION TITLE
Control Type with goField directive

### DISCUSSION THREAD

**Discussion by ChrisSargent** [99designs/gqlgen#1910] [Ideas]
Currently we have this directive....
`directive @goField(forceResolver: Boolean, name: String) on INPUT_FIELD_DEFINITION | FIELD_DEFINITION`

It would be useful if we could add a `type` parameter to be able to explicitly set the output Go type.

e.g.
`directive @goField(forceResolver: Boolean, name: String, type: String) on INPUT_FIELD_DEFINITION | FIELD_DEFINITION`

and can be used like 

```graphql
twoFoo: Int! @goField(type: "MyType")
```
outputting...
```go
twoFoo MyType
```

**Comment by ChrisSargent** [comment]
Or, if there's another way of achieving this currently then I'd be very happy to hear it.

**Comment by 3nvi** [comment]
I've made a proposal [here](https://github.com/99designs/gqlgen/issues/1620) that would help with what you're after :)

**Comment by ChrisSargent** [comment]
> I've made a proposal [here](https://github.com/99designs/gqlgen/issues/1620) that would help with what you're after :)

I did see that but I wasn't sure it was going to help our particular case.

```gql
type User {
  id: ID!
}

type Account {
  UserID: ID!
}
```

What I want to be output is:

```go
type User struct {
  id UserID
}

type Account struct {
  UserID UserID
}
```

**Comment by 3nvi** [comment]
With my proposal that would be achieved with the following:


```yaml
User:
  fields:
    id:
      fieldType: UserID

Account:
  fields:
    UserID:
      fieldType: UserID
```

The only caveat would be that non-native types would need to be imported (if not generated by gqlgen). If that's the case, then you'd most likely have to write: 

```yaml
fieldType: github.com/user/repo/../mypackage.UserID
```

**Comment by ChrisSargent** [comment]
ok, interesting. But, I think I'd feel like I was maintaining two schemas - i.e. if I change the graphql file, I'd also have to change the config file, which is really what I'm trying to avoid.

**Comment by 3nvi** [comment]
Got it. Gqlgen already has YAML as an alternative to `@goModel` for people that want to keep things separate (hence my proposal), but I do understand that this should also be an option as a GraphQL directive.

I'll let the gqlgen team handle the way forward :)

**Comment by razorness** [comment]
The current way is to manage your own struct in go and reference it in `gqlgen.yml` as model.
I have the case where I use a mixtures of (to string serializable) scalar types. For example ObjectID from Mongo and a custom string from PostgreSQL. I think it would be nice to have directive to reference a type explicitely.

**Comment by samlof** [comment]
My use case for this is with globally unique ids. Prefix type in id resolvers and use string in Node queries argument, everywhere else uuid with own marshaller since most places type is already known
Currently the doc on github readme has `There isn't any way around this, gqlgen has no way to know what you want in a given context.` so the directive could be a way to tell gqlgen what is wanted
